backend/graphql/service/product.service.ts

```diff
@@ -1,13 +1,73 @@
-import { ProductModel } from '../schema/product.schema'
-import { type GetProductInput } from '../types/product.type'
+import { type Product, ProductModel } from '../schema/product.schema'
+import {
+  type UpdateProductInput,
+  type CreateProductInput,
+  type GetProductInput
+} from '../types/product.type'
 import { NotFoundError } from '../errors/notFoundError'
+import { type Context } from '../types/context.type'
 
 export class ProductService {
-  findProducts() {
-    return ProductModel.find()
+  async createProduct(
+    input: CreateProductInput,
+    context: Context
+  ): Promise<Product> {
+    if (!context.user?.isAdmin) {
```

Comment:
Amazing!

But it's fancier if you have a decorator similar to `@Authorized` in the resolver that checks for the user role 😜